src/main/kotlin/org/rust/ide/intentions/SortImplTraitMethodsIntention.kt

```diff
@@ -0,0 +1,47 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.ide.intentions
+
+import com.intellij.openapi.editor.Editor
+import com.intellij.openapi.project.Project
+import com.intellij.psi.PsiElement
+import org.rust.lang.core.psi.RsImplItem
+import org.rust.lang.core.psi.ext.ancestorStrict
+import org.rust.lang.core.psi.ext.resolveToTrait
+
+class SortImplTraitMethodsIntention : RsElementBaseIntentionAction<RsImplItem>() {
+
+    override fun getText() = "Apply same method order"
+    override fun getFamilyName(): String = text
+
+    override fun findApplicableContext(project: Project, editor: Editor, element: PsiElement): RsImplItem? {
+        val implItem = element.ancestorStrict<RsImplItem>() ?: return null
+        if (!isApplicableTo(implItem)) return null
+        return implItem
+    }
+
+    override fun invoke(project: Project, editor: Editor, ctx: RsImplItem) {
+        val implMethods = ctx.members?.functionList ?: return
```

Comment:
Maybe sort not only functions but all members: functions, associated types and associated consts?